app/models/concerns/rdf/graphable/dereferenceable.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module RDF
+  module Graphable
+    # Dereference RDF statements if their value matches the provided 'only' regex.
+    # If no regex is provided it will only attempt to dereference values starting with "http://" or "https://".
+    #
+    # Runs during the +graph+ callback.
+    #
+    # @example
+    #   class MyDocument
+    #     include Mongoid::Document
+    #     include RDF::Graphable
+    #     include RDF::Graphable::Dereferenceable
+    #
+    #     field :dcterms_spatial, type: ArrayOf.type(String), default: []
+    #
+    #     dereferences RDF::Vocab::DC.spatial, only: %r(\Ahtttp://www.example.org/)
+    #   end
+    #
+    #   doc = MyDocument.new(dcterms_spatial: 'htttp://www.example.org/rdf_element)
+    #   doc.graph
+    #   doc.rdf_graph.query(predicate: RDF::Vocab::DC.spatial) #=> rdf
+    module Dereferenceable
+      extend ActiveSupport::Concern
+
+      class_methods do
+        def dereference_only_restrictions
+          @dereference_only_restrictions ||= {}
+        end
+
+        def dereferences(*predicates, **options)
+          only_restriction = options.delete(:only) || %r(\Ahttp(s)?://)
+          class_eval do
```

Comment:
Is this necessary given that this is a class method?

Reply:
doesn't appear to be. I've removed it in the literalisation and exclusion modules too.